src/components/TodoHeader/TodoHeader.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect, useRef, useState } from 'react';
+import classNames from 'classnames';
+
+type Props = {
+  isTempTodoTrue: boolean,
+  areAllTodosCompleted: boolean,
+  isToggleAllButtonVisible: boolean,
+  showErrorMessage: (message: string) => void,
+  handleEnterKeyPress: (todoTitle: string) => void,
+  onToggleAllButtonClick: () => void,
+};
+
+export const TodoHeader: React.FC<Props> = React.memo(({
+  isTempTodoTrue,
+  areAllTodosCompleted,
+  isToggleAllButtonVisible,
+  showErrorMessage,
+  handleEnterKeyPress,
+  onToggleAllButtonClick,
+}) => {
+  const [title, setTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    return () => {
+      if (inputRef.current) {
+        inputRef.current.focus();
+      }
+    };
+  }, [isTempTodoTrue]);
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputElement = event.target as HTMLInputElement;
+
+    setTitle(inputElement.value);
+  };
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
```suggestion
  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
```
https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364